### ISSUE TITLE
Long running test: ManyProducerConsumer_ConcurrentReadWrite_WithBufferedCapacity_Success

### ISSUE THREAD

**Issue by ViktorHofer** [dotnet/runtime#57941]
Configuration: `net6.0-Linux-Debug-x64-mono_interpreter_release-Debian.9.Amd64.Open`

```
  Discovering: System.Threading.Channels.Tests (method display = ClassAndMethod, method display options = None)
  Discovered:  System.Threading.Channels.Tests (found 454 test cases)
  Starting:    System.Threading.Channels.Tests (parallel test collections = on, max threads = 2)
    System.Threading.Channels.Tests.StressTests.CanceledReads [SKIP]
      Condition(s) not met: "IsStressModeEnabled"
    System.Threading.Channels.Tests.StressTests.ReadWriteVariations [SKIP]
      Condition(s) not met: "IsStressModeEnabled"
   System.Threading.Channels.Tests: [Long Running Test] 'System.Threading.Channels.Tests.BoundedChannelTests.ManyProducerConsumer_ConcurrentReadWrite_WithBufferedCapacity_Success', Elapsed: 00:02:22
   System.Threading.Channels.Tests: [Long Running Test] 'System.Threading.Channels.Tests.BoundedChannelTests.ManyProducerConsumer_ConcurrentReadWrite_WithBufferedCapacity_Success', Elapsed: 00:04:22
   System.Threading.Channels.Tests: [Long Running Test] 'System.Threading.Channels.Tests.BoundedChannelTests.ManyProducerConsumer_ConcurrentReadWrite_WithBufferedCapacity_Success', Elapsed: 00:06:22
   System.Threading.Channels.Tests: [Long Running Test] 'System.Threading.Channels.Tests.BoundedChannelTests.ManyProducerConsumer_ConcurrentReadWrite_WithBufferedCapacity_Success', Elapsed: 00:08:22
   System.Threading.Channels.Tests: [Long Running Test] 'System.Threading.Channels.Tests.BoundedChannelTests.ManyProducerConsumer_ConcurrentReadWrite_WithBufferedCapacity_Success', Elapsed: 00:10:22
   System.Threading.Channels.Tests: [Long Running Test] 'System.Threading.Channels.Tests.BoundedChannelTests.ManyProducerConsumer_ConcurrentReadWrite_WithBufferedCapacity_Success', Elapsed: 00:12:22
   System.Threading.Channels.Tests: [Long Running Test] 'System.Threading.Channels.Tests.BoundedChannelTests.ManyProducerConsumer_ConcurrentReadWrite_WithBufferedCapacity_Success', Elapsed: 00:14:22
['System.Threading.Channels.Tests' END OF WORK ITEM LOG: Command timed out, and was killed]
```

Console: https://helixre8s23ayyeko0k025g8.blob.core.windows.net/dotnet-runtime-refs-pull-57854-merge-219d9b12b2284beea4/System.Threading.Channels.Tests/1/console.a78ef497.log?sv=2019-07-07&se=2021-09-12T14%3A18%3A11Z&sr=c&sp=rl&sig=yqUs%2BjHW5hDM11pKterFAMjhHW4TumUDatb7oQQ7Ggo%3D

Build: https://dev.azure.com/dnceng/public/_build/results?buildId=1311705&view=ms.vss-test-web.build-test-results-tab&runId=38609488&paneView=debug

**Comment by danmoseley** [comment]
hit again https://github.com/dotnet/runtime/pull/67827/checks?check_run_id=5979100266
on net7.0-Linux-Debug-x64-mono_interpreter_release-(Debian.10.Amd64.Open)Ubuntu.1804.Amd64.Open@mcr.microsoft.com/dotnet-buildtools/prereqs:debian-10-helix-amd64-20210304164434-56c6673

@BrzVlad any theory why this is happening on interpreter? Not sure whether it's hung or just very slow.

**Comment by SamMonoRT** [comment]
@BrzVlad - any ideas, anything we should fix for 7.0.0?

**Comment by BrzVlad** [comment]
I'm still not able to reproduce this and we don't have any disabled tests for this issue. Closing this issue for now.
